apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/common/ClaimDao.java

```diff
@@ -68,6 +84,75 @@ public Object getEntityById(ResourceTypeV2<?> type, String id) {
     return claimEntity;
   }
 
+  public <T> List<T> findAllByMbiHash(
+      Class<T> entityClass, String mbiHash, DateRangeParam lastUpdated) {
+    return findAllByAttribute(entityClass, "mbiHash", mbiHash, lastUpdated);
+  }
+
+  public <T> List<T> findAllByMbi(Class<T> entityClass, String mbi, DateRangeParam lastUpdated) {
+    return findAllByAttribute(entityClass, "mbi", mbi, lastUpdated);
+  }
+
+  @VisibleForTesting
+  <T> List<T> findAllByAttribute(
+      Class<T> entityClass,
+      String attributeName,
+      String attributeValue,
+      DateRangeParam lastUpdated) {
+    List<T> claimEntities = null;
+
+    CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<T> criteria = builder.createQuery(entityClass);
+    Root<T> root = criteria.from(entityClass);
+
+    criteria.select(root);
+    criteria.where(
+        builder.and(
+            builder.equal(root.get(attributeName), attributeValue),
+            lastUpdated == null
+                ? builder.and()
+                : createDateRangePredicate(root.get("lastUpdated"), lastUpdated, builder)));
```

Comment:
what do you think about adding an index to the lastUpdated field?

Reply:
Personally I'd be fine with putting the code through without the index until we can benchmark with large amounts of data to determine exactly what we'd need.

Reply:
I'd also wait to see if any performance issues exist, as indexing the column will also have a negative hit on the the ETL process, which we're unsure how much traffic will be there atm.